File: src/server/Adfnet.Service/Models/MenuModel.cs
using System;
using Adfnet.Core;
using Adfnet.Core.ValueObjects;

namespace Adfnet.Service.Models
{
    public class MenuModel : IServiceModel
    {
        public Guid Id { get; set; }
        public int DisplayOrder { get; set; }
        public bool IsApproved { get; set; }
        public int Version { get; set; }
        public DateTime CreationTime { get; set; }
        public IdCodeName Creator { get; set; }
        public DateTime LastModificationTime { get; set; }
        public IdCodeName LastModifier { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Address { get; set; }
        public string Icon { get; set; }
        public IdCodeName ParentMenu { get; set; }


    }
}

File: src/client/web/private/src/app/models/menu-model.ts
import { IdCodeName } from '../value-objects/id-code-name';
import { IdName } from '../value-objects/id-name';

export class MenuModel {
    id: string;
    displayOrder: number;
    isApproved: boolean;
    version: number;
    creationTime: Date;
    creator: IdName;
    lastModificationTime: Date;
    lastModifier: IdName;
    code: string;
    name: string;
    description: string;
    address: string;
    icon: string;
    parentMenu: IdCodeName;
    constructor() {
        this.creator = new IdName();
        this.lastModifier = new IdName();
        this.parentMenu = new IdCodeName();
    }
}
